plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Escrow.hs

```diff
@@ -0,0 +1,285 @@
+{-# LANGUAGE TemplateHaskell   #-}
+{-# LANGUAGE TypeFamilies      #-}
+-- | An general-purpose escrow contract in Plutus
+module Language.PlutusTx.Coordination.Contracts.Escrow(
+    -- $escrow
+    Escrow
+    , EscrowParams(..)
+    , EscrowTarget(..)
+    , payToScriptTarget
+    , payToPubKeyTarget
+    , escrowAddress
+    , escrowScript
+    , targetValue
+    -- * Actions
+    , pay
+    , payEp
+    , redeem
+    , redeemEp
+    , refund
+    , refundEp
+    , RedeemFailReason(..)
+    , RedeemResult(..)
+    , RefundResult(..)
+    , EscrowSchema
+    ) where
+
+import           Control.Applicative            (Applicative (..))
+import           Control.Lens                   ((&), (^.), (.~), at, folded)
+import qualified Data.Set                       as Set
+import qualified Ledger
+import           Ledger                         (Address, DataScript(..), PubKey, Slot, ValidatorHash, scriptOutputsAt,
+                                                 txSignedBy, valuePaidTo, interval, TxId, TxOutRefOf(..), ValidatorScript, TxOut, pubKeyTxOut, scriptTxOut')
+import           Ledger.Interval                (after, before, from)
+import qualified Ledger.Interval                as Interval
+import           Ledger.AddressMap              (values)
+import qualified Ledger.Typed.Scripts           as Scripts
+import qualified Ledger.Scripts                 as Scripts
+import           Ledger.Validation              (PendingTx, PendingTx' (..))
+import           Ledger.Value                   (Value, lt)
+
+import qualified Language.Plutus.Contract.Typed.Tx as Typed
+import           Language.Plutus.Contract
+import qualified Language.PlutusTx              as PlutusTx
+import           Language.PlutusTx.Prelude      hiding (Applicative (..), check)
+
+import qualified Prelude                        as Haskell
+
+type EscrowSchema =
+    BlockchainActions
+        .\/ Endpoint "pay-escrow" (PubKey, Value)
+        .\/ Endpoint "redeem-escrow" ()
+        .\/ Endpoint "refund-escrow" ()
+
+-- $escrow
+-- The escrow contract implements the exchange of value between multiple 
+-- parties. It is defined by a list of targets (public keys and script 
+-- addresses, each associated with a value). It works similar to the 
+-- crowdfunding contract in that the contributions can be made independently,
+-- and the funds can be unlocked only by a transaction that pays the correct
+-- amount to each target. A refund is possible if the outputs locked by the 
+-- contract have not been spent by the deadline. (Compared to the crowdfunding
+-- contract, the refund policy is simpler because here because there is no
+-- "collection period" during which the outputs may be spent after the deadline
+-- has passed. This is because we're assuming that the participants in the escrow
+-- contract will make their deposits as quickly as possible after agreeing on a
+-- deal)
+-- 
+-- The contract supports two modes of operation, manual and automatic. In 
```

Comment:
Is this implemented?

Reply:
The endpoints are implemented, that's what is used in the tests. I'll add a contract that combines them all (just like the crowdfunding one)